doc/v5.0/man3/MPI_T_init_thread.3
.\" Automatically generated by Pandoc 2.14.2
.\"
.TH "MPI_T_init_thread" "3" "" "2021-08-30" "Open MPI"
.hy
.SH NAME
.PP
MPI_T_init_thread - Initializes the MPI Tool information interface
.SH SYNTAX
.SS C Syntax
.IP
.nf
\f[C]
#include <mpi.h>
int MPI_T_init_thread(int required, int *provided)
\f[R]
.fi
.SH INPUT PARAMETERS
.IP \[bu] 2
required: Desired level of thread support (integer).
.SH OUTPUT PARAMETERS
.IP \[bu] 2
provided: Available level of thread support (integer).
.SH DESCRIPTION
.PP
\f[C]MPI_T_init_thread()\f[R] initializes the MPI tool information
interface.
Calls to MPI tool functions are allowed at any point in execution
(including before \f[C]MPI_Init()\f[R] and after
\f[C]MPI_Finalize()\f[R]) as long as \f[C]MPI_T_init_thread()\f[R] has
been called at least once and the number of calls to
\f[C]MPI_T_init_thread()\f[R] is greater than the number of calls to
\f[C]MPI_T_finalize()\f[R].
If at any point in execution the number of calls to
\f[C]MPI_T_finalize()\f[R] equals the number of calls to
\f[C]MPI_T_init_thread()\f[R] the MPI tool interface will no longer be
available until another call to \f[C]MPI_T_init_thread()\f[R].
.PP
\f[C]MPI_T_init_thread()\f[R], like \f[C]MPI_Init_thread()\f[R], has a
provision to request a certain level of thread support in
\f[C]required\f[R]:
.IP \[bu] 2
\f[C]MPI_THREAD_SINGLE\f[R]: Only one thread will execute.
.IP \[bu] 2
\f[C]MPI_THREAD_FUNNELED\f[R]: If the process is multithreaded, only the
thread that called \f[C]MPI_Init_thread()\f[R] will make MPI calls.
.IP \[bu] 2
\f[C]MPI_THREAD_SERIALIZED\f[R]: If the process is multithreaded, only
one thread will make MPI library calls at one time.
.IP \[bu] 2
\f[C]MPI_THREAD_MULTIPLE\f[R]: If the process is multithreaded, multiple
threads may call MPI at once with no restrictions.
.PP
The level of thread support available to the program is set in
\f[C]provided\f[R].
In Open MPI, the value is dependent on how the library was configured
and built.
Note that there is no guarantee that \f[C]provided\f[R] will be greater
than or equal to \f[C]required\f[R].
.SH NOTES
.PP
It is the caller\[cq]s responsibility to check the value of
\f[C]provided\f[R], as it may be less than what was requested in
\f[C]required\f[R].
.SH ERRORS
.PP
\f[C]MPI_T_init_thread()\f[R] will fail if:
.IP \[bu] 2
\f[C]MPI_T_ERR_MEMORY\f[R]: Out of memory
.IP \[bu] 2
\f[C]MPI_T_ERR_CANNOT_INIT\f[R]: Interface not in the state to be
initialized
.SH SEE ALSO
.PP
\f[C]MPI_T\f[R](5), \f[C]MPI_Init_thread\f[R](3),
\f[C]MPI_T_finalize\f[R](3)


doc/v3.0/man3/MPI_T_init_thread.3.php
<?php
$topdir = "../../..";
$title = "MPI_T_init_thread(3) man page (version 3.0.6)";
$meta_desc = "Open MPI v3.0.6 man page: MPI_T_INIT_THREAD(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
      <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_T_init_thread</b> - Initializes the MPI Tool imformation interface

<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_T_init_thread(int required, int *provided)
</pre>
<h2><a name='sect3' href='#toc3'>Input Parameters</a></h2>

<dl>

<dt>required </dt>
<dd>Desired level of thread support (integer).   </dd>
</dl>

<h2><a name='sect4' href='#toc4'>Output
Parameters</a></h2>

<dl>

<dt>provided </dt>
<dd>Available level of thread support (integer).   </dd>
</dl>

<h2><a name='sect5' href='#toc5'>Description</a></h2>
MPI_T_init_thread()
initializes the MPI tool information interface. Calls to MPI tool functions
are allowed at any point in execution (including before <a href="../man3/MPI_Init.3.php">MPI_Init</a>() and
after <a href="../man3/MPI_Finalize.3.php">MPI_Finalize</a>()) as long as MPI_T_init_thread() has been called at
least once and the number of calls to MPI_T_init_thread() is greater than
the number of calls to <a href="../man3/MPI_T_finalize.3.php">MPI_T_finalize</a>(). If at any point in execution the
number of calls to <a href="../man3/MPI_T_finalize.3.php">MPI_T_finalize</a>() equals the number of calls to MPI_T_init_thread()
the MPI tool interface will no longer be available until another call to
MPI_T_init_thread().
<p> <p>
MPI_T_init_thread, like <a href="../man3/MPI_Init_thread.3.php">MPI_Init_thread</a>, has a provision
to request a certain level of thread support in <i>required</i>:
<dl>

<dt>MPI_THREAD_SINGLE
</dt>
<dd>Only one thread will execute. </dd>

<dt>MPI_THREAD_FUNNELED </dt>
<dd>If the process is multithreaded,
only the thread that called <a href="../man3/MPI_Init_thread.3.php">MPI_Init_thread</a> will make MPI calls. </dd>

<dt>MPI_THREAD_SERIALIZED
</dt>
<dd>If the process is multithreaded, only one thread will make MPI library
calls at one time. </dd>

<dt>MPI_THREAD_MULTIPLE </dt>
<dd>If the process is multithreaded,
multiple threads may call MPI at once with no restrictions.  </dd>
</dl>
<p>
The level of
thread support available to the program is set in <i>provided</i>. In Open MPI,
the value is dependent on how the library was configured and built. Note
that there is no guarantee that <i>provided</i> will be greater than or equal
to <i>required</i>.
<h2><a name='sect6' href='#toc6'>Notes</a></h2>
It is the caller&rsquo;s responsibility to check the value of
<i>provided</i>, as it may be less than what was requested in <i>required</i>.
<p>
<h2><a name='sect7' href='#toc7'>Errors</a></h2>
MPI_T_init_thread()
will fail if:
<dl>

<dt>[MPI_T_ERR_MEMORY] </dt>
<dd>Out of memory </dd>

<dt>[MPI_T_ERR_CANNOT_INIT]
</dt>
<dd>
<p>Interface not in the state to be initialized
<p> </dd>
</dl>

<h2><a name='sect8' href='#toc8'>See Also</a></h2>
<br>
<pre><a href="../man3/MPI_Init_thread.3.php">MPI_Init_thread</a>
<a href="../man3/MPI_T_finalize.3.php">MPI_T_finalize</a>

<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
